packages/indexer/native/src/lib.rs

```diff
@@ -430,32 +432,44 @@ declare_types! {
                 return cx.throw_error(format!("Error assembling script: {:?}", script.unwrap_err()));
             }
             let script = script.unwrap();
-            let prefix = if cx.argument::<JsNumber>(2)?.value() as u32 == 1 {
+            let script_type = cx.argument::<JsNumber>(2)?.value() as u32;
+            let prefix = if script_type == 1 {
                 KeyPrefix::TxTypeScript
             } else {
                 KeyPrefix::TxLockScript
             };
             let mut start_key = vec![prefix as u8];
             start_key.extend_from_slice(&script.as_slice()[SCRIPT_SERIALIZE_OFFSET..]);
+
+            let io_type= cx.argument::<JsValue>(3)?;
+            let io_type_mark =
+            if io_type.is_a::<JsNumber>() {
+                let io_type_number = cx.argument::<JsNumber>(3)?.value() as u8;
+                vec![io_type_number]
+            } else {
+                vec![]
+            };
             let mut end_key = start_key.clone();
-            let from_block = cx.argument::<JsValue>(3)?;
+            let from_block = cx.argument::<JsValue>(4)?;
             if from_block.is_a::<JsNumber>() {
                 let from_block_number = from_block.downcast::<JsNumber>().or_throw(&mut cx)?.value() as u64;
                 start_key.extend_from_slice(&from_block_number.to_be_bytes());
             }
-            let to_block = cx.argument::<JsValue>(4)?;
+            let to_block = cx.argument::<JsValue>(5)?;
             if to_block.is_a::<JsNumber>() {
                 // here set to_block_number as toBlock + 1, making the toBlock included in query range.
                 let to_block_number = to_block.downcast::<JsNumber>().or_throw(&mut cx)?.value() as u64 + 1;
                 end_key.extend_from_slice(&to_block_number.to_be_bytes());
             } else {
                 end_key.extend_from_slice(&u64::MAX.to_be_bytes());
             }
+
```

Comment:
Remove this.

Reply:
This empty line is added here(and maybe somewhere else) for ergonomics reasons when I am looking at the long and crowded method body.